### PR TITLE
fix: cargo test fail in Sonoma or later of macos

### PR DESCRIPTION
**Pull request by Yougigun**
**Summary:**  
_Briefly describe the changes made in this PR._
potential reason:
https://github.com/alexcrichton/curl-rust/issues/524#issuecomment-1702156654

I followed this sulution.
https://github.com/alexcrichton/curl-rust/issues/524#issuecomment-1703325064

ref:
https://github.com/curl/curl/commit/6ab7e19
https://github.com/rust-lang/cargo/issues/12670
https://github.com/alexcrichton/curl-rust/issues/524

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh` to address and fix linting issues.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -49,6 +49,7 @@ nom = "7.1.3"
 exitcode = "1.1.2"
 log = "0.4.20"
 env_logger = "0.10.0"
+curl-sys = { version = "0.4", features = ["force-system-lib-on-osx"] }
```
[the remark this answers is not in the document]
Comment (Yougigun) [reply Cargo.toml]: done

**Review by tusharmath — APPROVED** [review]

**Comment by SomeoneToIgnore** [comment]
👋 
New curl-sys with the fix had been released recently, so this PR is hopefully not needed anymore.

**Comment by tusharmath** [comment]
@SomeoneToIgnore Ah! What timing. I just merged :D

**Comment by SomeoneToIgnore** [comment]
Oh, took me some time to fight the hesitation, sorry 🙂 
Either way, it's fixed now and you're aware of both ways, which is good.

**Comment by Yougigun** [comment]
to @SomeoneToIgnore 
thanks for the clarification. After updating curl-sys, it seems to work fine now. 
to @tusharmath 
in my subsequent PR, I will remove the force curl-sys deps, instead, update it in the lock file.

**Pull request merged**